app/controllers/catalog_controller.rb

```diff
@@ -198,4 +198,16 @@ def lookup_manifest
       render json: { message: "No manifest found for #{ark}" }, status: 404
     end
   end
+
+  def pdf
+    _, @document = fetch(params[:solr_document_id])
+    authorize! :show, resource
```

Comment:
I'm not sure if this is the right authorization to have. For example, to view the Figgy-generated PDF you only need `:pdf` permission, which isn't true for anonymous users if there's no pdf color-scheme set. For the full res PDF, it's more about the ability to `:download," which is super important for CDL. This is kind of a bummer, because it means you have to vary permission after you figure out which PDF to show them - and if they're not allowed to get the original PDF, can they get the Figgy generated PDF?

Reply:
👍🏻  I loaded the source_pdf and used `can?` to check permissions more specifically, so it can fall back on the generated pdf they are not allowed to download the original.